src/v0.8/core/collateral/base/escrow.sol

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+ *   (c) 2023 Dataswap
+ *
+ *  Licensed under the GNU General Public License, Version 3.0 or later (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.gnu.org/licenses/gpl-3.0.en.html
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.21;
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// upgrade
+import {RolesModifiers} from "src/v0.8/shared/modifiers/RolesModifiers.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+// types
+import {RolesType} from "src/v0.8/types/RolesType.sol";
+import {DatasetType} from "src/v0.8/types/DatasetType.sol";
+import {CollateralType} from "src/v0.8/types/CollateralType.sol";
+import {IDatasets} from "src/v0.8/interfaces/module/IDatasets.sol";
+
+// shared
+import {CollateralEvents} from "src/v0.8/shared/events/CollateralEvents.sol";
+
+/// @title Escrow
+/// @dev Base escrow contract, holds funds designated for a payee until they withdraw them.
+contract Escrow is
```

Comment:
It is necessary to improve other business contracts such as the dataset to associate business with staking

Reply:
Business logic will be submitted in other PR